src/host/VtInputThread.cpp

```diff
@@ -145,13 +165,16 @@ void VtInputThread::DoReadInput(const bool throwOnFail)
 //      have caused us to exit.
 DWORD VtInputThread::_InputThread()
 {
-    while (!_exitRequested)
+    auto onExitTriggerShutdown = wil::scope_exit([&] {
+        _shutdownEvent.SetEvent();
+    });
+
+    while (true)
     {
-        DoReadInput(true);
+        RETURN_IF_FAILED(_ReadInput());

```

Comment:
Ah. When the actual VtInputThread is started for reals, it will store its thread ID into _dwThreadId permanently and no one from the outside can use the public call anymore. I'll clarify in comments.